### PR TITLE
Solution (Spring Boot Data JPA).

### PR DESCRIPTION
**Pull request by maksym-bakunov**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +36,15 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
+
+    @PutMapping("/{id}")
+    public BookDto putBook(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/controller/BookController.java]:
```suggestion
    public BookDto update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 2]
```diff
@@ -13,10 +13,14 @@
...
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Getter
 @Setter
 @EqualsAndHashCode(exclude = "id")
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
 @NoArgsConstructor
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -40,4 +44,6 @@ public class Book {
     private String description;
     @Column(name = "cover_image")
     private String coverImage;
+    @Column(name = "is_deleted")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java]: remove all @Column(name..) hibernate automatically creates column in snake_case format

File: src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -36,4 +36,20 @@ public List<BookDto> findAll() {
...
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
+        return bookRepository.findById(id).map(b -> {
+            Book book = bookMapper.toModel(createBookRequestDto);
+            book.setId(b.getId());
+            bookRepository.save(book);
+
+            return bookMapper.toDto(book);
+        }).orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(()...
        bookMapper.updateBookFromDto(bookDto, book);
        bookRepository.save(book);
        return bookMapper.toDto(book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**